File: src/java/test/org/nesjs/nestest/Nestest.java
package org.nesjs.nestest;

import org.nesjs.core.*;

public class Nestest
{
    public static void main(String[] args) throws Exception
    {
        ROM.initMemoryWithFile("nestest.nes");
        
        MOS6502 _6502 = new MOS6502();
        
        _6502.init();
        _6502.reset();
        
        _6502.pc = 0xC000;
        
        _6502.step();
    }
}

File: nestest.js
/*
 * Runs the nestest.nes ROM and compares the output with a known good log.
 */

require("./ROM");
require("./Opcodes");
require("./Memory");
require("./6502");

ROM.loadFile("./nestest.nes");

CPU.init();
CPU.reset();

CPU.context.pc = 0xC000;

for(var _index = 0; _index < 2; _index++)
{
	CPU.step();
}

